### PR TITLE
Enhancement: `application_gateway` - `affinity_cookie_name`

### PR DESCRIPTION
**Pull request by mbfrahry**
Addressing one of the points of #1576
```
--- PASS: TestAccAzureRMApplicationGateway_cookieAffinity (1216.50s)
--- PASS: TestAccAzureRMApplicationGateway_basic (1156.14s)
```

### CODE REVIEW

File: azurerm/resource_arm_application_gateway.go [hunk 1]
```diff
@@ -1527,6 +1533,11 @@ func expandApplicationGatewayBackendHTTPSettings(d *schema.ResourceData, gateway
 			setting.ApplicationGatewayBackendHTTPSettingsPropertiesFormat.HostName = utils.String(hostName)
 		}
 
+		affinityCookieName := v["affinity_cookie_name"].(string)
+		if affinityCookieName != "" {
+			setting.AffinityCookieName = utils.String(affinityCookieName)
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_application_gateway.go]: do we not always want to assign this to ensure the value can be removed?

File: azurerm/resource_arm_application_gateway_test.go [hunk 2]
```diff
@@ -741,6 +741,31 @@ func TestAccAzureRMApplicationGateway_webApplicationFirewall_exclusions(t *testi
...
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMApplicationGatewayDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMApplicationGateway_cookieAffinity(ri, testLocation()),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMApplicationGatewayExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "backend_http_settings.0.affinity_cookie_name", "testCookieName"),
+				),
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_application_gateway_test.go]: could we also run a step to ensure this value can be removed?

File: azurerm/resource_arm_application_gateway.go (hunk 1 above)
Comment (mbfrahry) [reply azurerm/resource_arm_application_gateway.go]: It looks like it gets removed when you don't pass it in so we're good the way it is!

File: azurerm/resource_arm_application_gateway_test.go (hunk 2 above)
Comment (mbfrahry) [reply azurerm/resource_arm_application_gateway_test.go]: Added! 
```
--- PASS: TestAccAzureRMApplicationGateway_cookieAffinity (1239.32s)
```

**Comment by mbfrahry** [comment]
Addressing one of the points of #1576

**Pull request merged**